association-backend/src/webhooks/handlers/pretix/pretix_event_order_paid.py

```diff
@@ -0,0 +1,161 @@
+import logging
+from datetime import datetime, timezone
+from decimal import Decimal
+
+from dateutil import parser
+from dateutil.relativedelta import relativedelta
+from pythonit_toolkit.service_client import ServiceClient
+
+from src.association.settings import SERVICE_TO_SERVICE_SECRET, USERS_SERVICE_URL
+from src.association_membership.domain.entities import PaymentStatus, PretixPayment
+from src.association_membership.domain.repository import AssociationMembershipRepository
+from src.webhooks.exceptions import NoUserFoundWithEmail
+
+from .api import PretixAPI
+
+logger = logging.getLogger(__file__)
+
+GET_USER_ID_BY_EMAIL = """query UserIdByEmail($email: String!) {
+    userByEmail(email: $email) {
+        id
+    }
+}"""
+
+
+async def pretix_event_order_paid(payload):
+    action = payload["action"]
+    organizer = payload["organizer"]
+    event = payload["event"]
+    order_code = payload["code"]
+
+    pretix_api = PretixAPI(organizer, event)
+
+    order_data = pretix_api.get_order_data(order_code)
+    categories = {
+        result["id"]: result for result in pretix_api.get_categories()["results"]
+    }
+
+    order_positions = order_data["positions"]
+    membership_positions = []
+    for position in order_positions:
+        item_data = pretix_api.get_item_data(position["item"])
+        category = categories.get(item_data["category"], None)
+
+        assert (
```

Comment:
We are expecting to receive an error? For the other items of the order that are for the event and not for the association, I guess... ? 
How this is supposed to work?

Reply:
This checks that we found a category from the ones we fetched

I actually changed this code so avoid fetching N+1 items, now we fetch all the items of the category so we can just do a check if the item ID exists or not in this list